From cc3b8684e53641f22da8374148d99c762b6f877e Mon Sep 17 00:00:00 2001
From: Yao Xiao <xiaoyao@rock-chips.com>
Date: Mon, 19 Oct 2020 10:45:45 +0800
Subject: [PATCH] net: rockchip_wlan: bcmdhd_indep_power

1. add ENABLE_INSMOD_NO_FW_LOAD for init stability
2. reduce print level
3. modify module name for keepalive

Signed-off-by: Yao Xiao <xiaoyao@rock-chips.com>
Change-Id: I1be21817e53558945a35389ee02d8229e2b1a58a
---
 .../rkwifi/bcmdhd_indep_power/Makefile            |  6 +++---
 .../rkwifi/bcmdhd_indep_power/dhd_linux.c         | 15 +++++++--------
 2 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/Makefile b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/Makefile
index 100bcdf29c88..161880bd855d 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/Makefile
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 # bcmdhd
 
-MODULE_NAME := bcmdhd
+MODULE_NAME := bcmdhd_indep_power
 #CONFIG_BCMDHD := m
 CONFIG_BCMDHD_SDIO := y
 #CONFIG_BCMDHD_PCIE := y
@@ -26,7 +26,7 @@ DHDCFLAGS = -Wall -Wstrict-prototypes -Dlinux -DBCMDRIVER                 \
 	-DDHD_DONOT_FORWARD_BCMEVENT_AS_NETWORK_PKT                           \
 	-DMULTIPLE_SUPPLICANT -DTSQ_MULTIPLIER -DMFP -DDHD_8021X_DUMP         \
 	-DWL_EXT_IAPSTA -DWL_ESCAN -DCCODE_LIST                               \
-	-DDHD_UNSUPPORT_IF_CNTS                    \
+	-DDHD_UNSUPPORT_IF_CNTS -DENABLE_INSMOD_NO_FW_LOAD                    \
 	-Idrivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power \
 	-Idrivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/include
 
@@ -148,7 +148,7 @@ ifeq ($(CONFIG_LOAD_CHIPALIVE),y)
 endif
 
 #obj-$(CONFIG_RKWIFI) += $(MODULE_NAME).o
-obj-$(CONFIG_AP6XXX_INDEP_POWER) += bcmdhd.o
+obj-$(CONFIG_AP6XXX_INDEP_POWER) += bcmdhd_indep_power.o
 $(MODULE_NAME)-objs += $(DHDOFILES)
 
 ifeq ($(CONFIG_MACH_PLATFORM),y)
diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/dhd_linux.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/dhd_linux.c
index 8f0221b69adb..8bfd439e96b9 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/dhd_linux.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd_indep_power/dhd_linux.c
@@ -13702,7 +13702,6 @@ static int tcp_param_show(struct seq_file *s, void *data)
 {
 	seq_printf(s, "dhd_priv wl tcpka_conn_add 1 %s %s %s 1 %d %d 1 1 1 1 1 2 0xc000\n",
 			deabuf, sabuf, dabuf, source, dest);
-	pr_err("haha\n");
 
 	return 0;
 }
@@ -19172,15 +19171,15 @@ dhd_tcp_dump(char *ifname, uint8 *pktdata, bool tx)
 			switch (opt) {
 			case 0x00:
 				/* End of options. */
-				pr_err("rk tcp_parseopt: EOL\n");
+				pr_debug("rk tcp_parseopt: EOL\n");
 				return;
 			case 0x01:
 				/* NOP option. */
 				++c;
-				pr_err("rk tcp_parseopt: NOP\n");
+				pr_debug("rk tcp_parseopt: NOP\n");
 				break;
 			case 0x02:
-				pr_err("rk tcp_parseopt: MSS\n");
+				pr_debug("rk tcp_parseopt: MSS\n");
 				if (opts[c + 1] != 0x04 || c + 0x04 > max_c) {
 					/* Bad length */
 					pr_err("rk tcp_parseopt: bad length\n");
@@ -19190,7 +19189,7 @@ dhd_tcp_dump(char *ifname, uint8 *pktdata, bool tx)
 				c += 0x04;
 				break;
 			case 0x08:
-				pr_err("rk tcp_parseopt: TS\n");
+				pr_debug("rk tcp_parseopt: TS\n");
 				if (opts[c + 1] != 0x0A || c + 0x0A > max_c) {
 				  /* Bad length */
 				  pr_err("rk tcp_parseopt: bad length\n");
@@ -19213,7 +19212,7 @@ dhd_tcp_dump(char *ifname, uint8 *pktdata, bool tx)
 				c += 0x0A;
 				break;
 			default:
-				pr_err("rk tcp_parseopt: other\n");
+				pr_debug("rk tcp_parseopt: other\n");
 				if (opts[c + 1] == 0) {
 				  pr_err("rk tcp_parseopt: bad length\n");
 				  /* If the length field is zero, the options are malformed
@@ -19228,9 +19227,9 @@ dhd_tcp_dump(char *ifname, uint8 *pktdata, bool tx)
 	}
 
 	if (tx)
-		DHD_ERROR(("[%s-%s] %s %s %s %s %d, %d, %d, %8u, %8u, %d, %d, %d, %8u, %8u\n", ifname, tx ? "tx" : "rx",
+		pr_debug("[%s-%s] %s %s %s %s %d, %d, %d, %8u, %8u, %d, %d, %d, %8u, %8u\n", ifname, tx ? "tx" : "rx",
 					sabuf, dabuf, seabuf, deabuf, ip_id, source, dest, seq, seq_ack, window,
-					tcp_option_len, tcp_data_len, tsval, tsecr));
+					tcp_option_len, tcp_data_len, tsval, tsecr);
 }
 
 #ifdef DHD_DHCP_DUMP
-- 
2.35.3

